js/ui/payg.js

```diff
@@ -395,116 +484,214 @@ var PaygUnlockWidget = GObject.registerClass({
                 x_align: Clutter.ActorAlign.CENTER,
             });
 
-            entrySpinnerBox.add_child(suffix);
+            this._codeEntryLayout.add_child(suffix);
         }
 
-        this._buttonBox = new St.BoxLayout({
-            style_class: 'notification-actions',
-            x_expand: true,
-            vertical: true,
-        });
-        global.focus_manager.add_group(this._buttonBox);
-        this._buttonBox.add_child(entrySpinnerBox);
-
-        this._applyButton = this._createApplyButton();
-        this._applyButton.connect('clicked', this.startVerifyingCode.bind(this));
-        this._buttonBox.add_child(this._applyButton);
+        this.contentLayout.add_child(this._codeEntryLayout);
 
-        this.updateSensitivity();
-    }
+        /* Add buttons */
+        this._closeButton = this.addButton({
+            label: _('Close'),
+            key: Clutter.KEY_Escape,
+            action: () => this.close(),
+        });
 
-    _createCodeEntry() {
-        const codeEntry = new PaygUnlockCodeEntry({
-            style_class: 'notification-payg-entry',
-            x_expand: true,
-            can_focus: true,
+        this._applyButton = this.addButton({
+            label: _('Add Credit'),
```

Comment:
I would give this `key: Clutter.KEY_Enter` (or if there's a way to make it the default action for hitting Enter in the field, that instead)

Reply:
Good suggestion.  However, **_applyButton** type of **PaygUnlockCodeEntry** has already handled it.  And, **_applyButton**'s `activate` signal has been bound with `_apply()`.